### PR TITLE
Feature/pm 308 velocity using incremental encoder

### PR DESCRIPTION
**Pull request by Roelemans**
Closes PM-308

## Description

Uses the incremental encoder to determine velocity for the linear joints. This has a much higher resolution allowing less smoothing and a higher accuracy.

### CODE REVIEW

File: march_hardware/include/march_hardware/IMotionCube.h [hunk 1]
```diff
@@ -83,6 +84,7 @@ class IMotionCube : public Slave
   AbsoluteEncoder absolute_encoder_;
   IncrementalEncoder incremental_encoder_;
   ActuationMode actuation_mode_;
+  bool incremental_more_precise_;
```
Comment (Joris-JW) [inline march_hardware/include/march_hardware/IMotionCube.h]:
```suggestion
  bool is_incremental_more_precise_;
```

**Review by Joris-JW — COMMENTED** [review]
Have u looked into a way of removing the noise of the incremental encoder data line of the rotational joint. If this is indeed more precise it maybe useful? We are probably not the first!

**Review by RPKrijnen — COMMENTED** [review]
I fear that over time the incremental encoder might become less precise than the absolute encoder. This would be caused by backlash in the motor (however slight). It could be useful to continuously compare the value of the incremental encoder with the absolute encoder. 
The absolute encoder does not experience backlash and is therefore capable of providing a reference signal. Frequent enough updates can cause the drift of the signal to minimize while still maintaining high precision.

**Comment by RutgerVanBeek** [comment]
> I fear that over time the incremental encoder might become less precise than the absolute encoder. This would be caused by backlash in the motor (however slight). It could be useful to continuously compare the value of the incremental encoder with the absolute encoder.
> The absolute encoder does not experience backlash and is therefore capable of providing a reference signal. Frequent enough updates can cause the drift of the signal to minimize while still maintaining high precision.

You are right, therefore we decided to start of easy and use the incremental encoder for velocity calculations only. Then we should not have the drifting problem. In the future we might also choose to use the incremental encoder for the position, then indeed frequent updates seem necessary.

**Comment by Roelemans** [comment]
> Have u looked into a way of removing the noise of the incremental encoder data line of the rotational joint. If this is indeed more precise it maybe useful? We are probably not the first!

Certainly. You can probably combine the incremental encoder with the absolute for a much better signal. This will require some research though.

**Review by Joris-JW — DISMISSED** [review]
yeeeet

**Review by Olavhaasie — APPROVED** [review]

**Review by RutgerVanBeek — APPROVED** [review]

**Pull request merged**